Documentation/dev/deploy-metering.md

```diff
@@ -0,0 +1,165 @@
+# Overview
+
+The `deploy-metering` utility emulates the install/uninstall logic that's currently done in the `hack/install.sh` and`hack/uninstall.sh` shell scripts.
+
+When interacting with the `deploy-metering` binary, you have the choice of using the command flags available, or their respective environment variable.
+
+In the case where both the flag, and the environment variable are specified, the [flag value takes precedence](#example-of-precedence).
+
+## Building the binary
+
+```bash
+make bin/deploy-metering
+```
+
+### Prerequisites
+
+With the addition of the deploy-metering CLI, we can assign a default (typically an empty value) value for most of the configuration options.
+
+This means the only required flags/environment variables that needs to be set, is the namespace in which the Metering stack will be managed and path to the `MeteringConfig` custom resource file.
+
+Before deploying metering, ensure the either of the following options are set:
+
+```bash
+export METERING_NAMESPACE="metering"
+export METERING_CR_FILE=<path to custom MeteringConfig custom resource>
+```
+
+Or, to run the CLI using the corresponding flags:
+
+```bash
+./bin/deploy-metering install --namespace metering --meteringconfig <path to custom MeteringConfig custom resource>
+```
+
+It's important to note that when the `MeteringConfig` custom resource is unspecified, the package uses the default `MeteringConfig` manifest resource, which contains an invalid storage specification.
+
+If you do not set that environment variable or flag, the installation will succeed, but it will fail the validation check in the metering-ansible-operator Role due to an invalid storage specification.
+
+### General Configuration Options
```

Comment:
Could add that KUBECONFIG is also an env var that needs to be set. Not sure how much basic Kubernetes understanding we are catering to here.

Reply:
@EmilyM1 that's a `oc` thing, so it doesn't apply here.

Reply:
@pruan-rht I think this may be a valid point, but let's hold off for now on listing the more granule requirements as pre-requisites until we flesh this initial documentation more.

Reply:
This is also why I threw this in the dev documentation for now, until there's a better avenue with interacting with this tool to expose more generally.